### PR TITLE
SuperH: simplify div1 logic

### PR DESCRIPTION
**Pull request by VGKintsugi**
asdf-prime's modifications to div1 instruction. I don't have the skill to verify this is correct. See PR: #2478

### CODE REVIEW

File: Ghidra/Processors/SuperH/data/languages/superh.sinc [hunk 1]
```diff
@@ -1021,74 +1021,25 @@ MovMUReg2: MovMUReg2_15 is MovMUReg2_15 {
...
+    m_eq_q = $(M_FLAG) == $(Q_FLAG);
     
-    # FIXME: cleaner way to do this??
     tmp0 = rn_08_11;
+    # rn_08_11 = old_q_eq_m ? rn_08_11 - tmp2 : rn_08_11 + tmp2;
+    rn_08_11 = (zext(old_q_eq_m) * (rn_08_11 - tmp2)) + (zext(!old_q_eq_m) * (rn_08_11 + tmp2));
+    # tmp1 = old_q_eq_m ? rn_08_11 > tmp0 : rn_08_11 < tmp0;
+    tmp1 = (old_q_eq_m * (rn_08_11 > tmp0)) + (!old_q_eq_m * (rn_08_11 < tmp0));
+    # $(Q_FLAG) = m_eq_q ? tmp1 : tmp1 == 0;
+    $(Q_FLAG) = (m_eq_q * tmp1) + (!m_eq_q & (tmp1 == 0));
```
Comment (ghidra1) [inline Ghidra/Processors/SuperH/data/languages/superh.sinc:1041]: Re: `$(Q_FLAG) = (m_eq_q * tmp1) + (!m_eq_q & (tmp1 == 0));`
  Use * not &

File: Ghidra/Processors/SuperH/data/languages/superh.sinc [hunk 2]
```diff
@@ -1021,74 +1021,25 @@ MovMUReg2: MovMUReg2_15 is MovMUReg2_15 {
     local tmp1:1;
     local tmp2:4;
     local old_q:1;
+    local old_q_eq_m:1;
```
Comment (ghidra1) [inline Ghidra/Processors/SuperH/data/languages/superh.sinc:1024]: Not sure why local vars are pre-declared (although this had previously been done this way).  I generally find declaration on first write to be easier to read. (OPTIONAL)

**Review by ghidra1 — CHANGES_REQUESTED** [review]

**Comment by VGKintsugi** [comment]
The choosery pattern change has been implemented. I skipped the optional change for fear of breaking things. I think this PR should be handled at the same time: https://github.com/NationalSecurityAgency/ghidra/pull/2863 because I mixed up the delay slots fix with this PR as well.

**Comment by VGKintsugi** [comment]
@ryanmkurtz @ghidra1, @GhidorahRex,

Something with this merge isn't making sense to me. I'm not seeing any changes to the div1 instruction? The referenced commit (https://github.com/NationalSecurityAgency/ghidra/commit/e21149816cc4469b2f824bbba74692431456c486) appears to include the fixes for the delay slot instructions, but not the changes to div1. I'm assuming this is an error?

**Comment by ryanmkurtz** [comment]
Yes, something went wrong on our end.   We are looking into it.

**Comment by VGKintsugi** [comment]
@ryanmkurtz, 

I realize you guys are busy but can we try to get this PR into Ghidra 10.0? As is the Ghidra 10.0 Beta release notes (https://ghidra-sre.org/releaseNotes_10.0beta.html) are incorrect:  "Processors. Simplified SuperH div1 instruction. (GP-753, Issue #2864)". 

So we can update the release notes or get this PR out. Let me know if there's anything I can do to expedite the process. Thanks.

**Comment by ryanmkurtz** [comment]
We left in the release note because we planned on getting it in for 10.0-final.  I temporarily removed the 10.0 milestone because it's still in internal review.  If it doesn't make it in for whatever reason, we plan on removing the release note.

**Comment by ghidra1** [comment]
Change has been merged

**Pull request closed without merging**